js.go

```diff
@@ -2453,6 +2453,9 @@ func checkMsg(msg *Msg, checkSts bool) (usrMsg bool, err error) {
 		// implementation. Regardless, ignore 408 errors until receiving at least
 		// one message.
 		err = ErrTimeout
+	case leadershipChangedSts:
+		// 409 indicates that the consumer leadership changed no the server. The client should ignore 409 errors.
+		err = ErrTimeout
```

Comment:
We are revisiting the handling of the ErrTimeout in Fetch at the moment, so not sure if this makes sense here. If the leadership changed while the library was waiting to get a message (but did not get any), your change would cause Fetch() to go back to sub.nextMsgWithContext() without resending a new request, which means that the library would just sit in the sub's next until timeout and would never receive a message even if some became available. This error from the server means that the request is invalid, so a new request need to be sent.